### PR TITLE
Terms: move TermTreeSelector, add devdocs example.

### PR DESCRIPTION
**Pull request by timmyc**
Suggested in #5366 - this branch moves the `<TermTreeSelector />` to the components directory, and adds an example to devdocs.  Additionally while making this move, I moved away from getting the `siteId` via `getSelectedSiteId` in `term-tree-selector/terms.jsx`, and actually switched over to using the prop ( as the README says you should ).

This was necessary to make the devdocs example work, but also seems like the precedent used in the `PostSelector`, and it also fixes an issue reported by @aduth where an extra request happens to `/site/{ site }/terms` when switching sites in the menu builder.

__To Test__
- Open [menus](http://calypso.localhost:3000/menus/) for a site, validate you can set/update a menu target using the category accordion
- View the new [TermTreeSelector Devdocs example](http://calypso.localhost:3000/devdocs/app-components/term-tree-selector)

Test live: https://calypso.live/?branch=move/term-tree-selector

### CODE REVIEW

File: client/components/term-tree-selector/terms.jsx [hunk 1]
```diff
@@ -362,7 +361,7 @@ const TermTreeSelectorList = React.createClass( {
 
 		return (
 			<div ref={ this.setSelectorRef } className={ classes }>
-				{ this.state.requestedPages.map( ( page ) => (
+				{ siteId && this.state.requestedPages.map( ( page ) => (
```
Comment (aduth) [inline client/components/term-tree-selector/terms.jsx]: I think we'll need to update `componentWillReceiveProps` to reset the default state when `siteId` changes.

Comment (timmyc) [inline client/components/term-tree-selector/terms.jsx]: Excellent point - I will change that up

**Comment by timmyc** [comment]
@aduth fixed up `componentWillReceiveProps`

**Comment by aduth** [comment]
@mtias mentioned that until we have a proper directory for "App Components" ([distinct from visual components](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#app-components)), maybe we shouldn't move this from where it's at.

Probably a pain to undo since it's all in a single commit 😧 

Ref: p4TIVU-3B9-p2 (thoughts on having a separate directory for app components)

**Comment by timmyc** [comment]
Was easier just to do a clean start in #6659.  Closing this.

**Pull request closed without merging**